packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts

```diff
@@ -0,0 +1,92 @@
+/**
+Copyright 2021 Forestry.io Holdings, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Cloudinary } from 'cloudinary-core'
+import {
+  Media,
+  MediaList,
+  MediaListOptions,
+  MediaStore,
+  MediaUploadOptions,
+} from '@tinacms/core'
+
+export class CloudinaryMediaStore implements MediaStore {
+  accept = '*'
+  //@ts-ignore
+  private api: Cloudinary
+
+  constructor(public cloud_name: string) {
+    this.api = new Cloudinary({
+      cloud_name: this.cloud_name,
+      secure: true,
+    })
+  }
+
+  async persist(media: MediaUploadOptions[]): Promise<Media[]> {
+    // const url = `https://api.cloudinary.com/v1_1/${process.env.NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME}/image/upload`
+
+    // TODO: Fix only one file upload
+    const { file, directory } = media[0]
+    const formData = new FormData()
+
+    formData.append('file', file)
+    formData.append('directory', directory)
+    formData.append('filename', file.name)
+
+    const res = await fetch(`/api/cloudinary/media`, {
```

Comment:
Would probably be better to move the actual Cloudinary API calls into code outside of the media store class, and/or into a `CloudinaryClient` class, for more modular code, and for consistency with the other media stores.